### ISSUE TITLE
Pyenv segment significantly slows down zsh

### ISSUE THREAD

**Issue by Zeta611** [bhilburn/powerlevel9k#994]
#### Describe Your Issue
Adding pyenv to a segment lags the prompt ~1 sec. on macOS.
`POWERLEVEL9K_LEFT_PROMPT_ELEMENTS=(context dir vcs pyenv)`
is the elements that I use on left prompt.

#### Have you tried to debug or fix it?
Removing pyenv from the left prompt fixes the issue.

#### Environment Information
This information will help us understand your configuration. 

  - zsh version: zsh 5.6.2
  - zplug 2.4.2
  - zplug
  - P9k v0.6.6
  - iTerm2 Build 3.2.20180825 on macOS 10.14

**Comment by dritter** [comment]
Thanks for the report @Zeta611 . I'm afraid, we can't do anything about that, unless we find a better way to retrieve the environment string, which we get by invoking pyenv. Do you know a better way?

**Comment by Zeta611** [comment]
If `cat` is faster than invoking `pyenv version-name`, reading from file `.python-version` maybe an alternative. Then
```
pyenv version-file-read ${pyenv_root}/version)
```
can also be replaced with `cat ${pyenv_root}/version`.
I'm not sure if this approach would work for everyone's environment, though.

**Comment by Syphdias** [comment]
I tried to reproduce it on Linux using P9K tag v0.6.6, zsh 5.6.2 in tilix and could barely notice the difference:
![image](https://user-images.githubusercontent.com/16988672/48585968-a7560500-e92e-11e8-8b30-bbc9261b21b1.png)
![image](https://user-images.githubusercontent.com/16988672/48586113-467afc80-e92f-11e8-9b90-1a0ecf948383.png)
![image](https://user-images.githubusercontent.com/16988672/48586276-c7d28f00-e92f-11e8-89cb-290c48e35e80.png)

@Zeta611, can you provide more details to how you use pyenv? How did you install it? brew, manually? Which python versions are installed and did you use any flags to compile them? Does it happen with all versions? Does it only happen if activated? Only if version is set by file or only if version is set by environment variable? Do you use multiple versions at the same time?

I have never used used pyenv before, so I'm not very familiar with it. Right now it look to me like it uses `$PYENV_VERSION`, otherwise the file `.python-version`. Also `pyenv version-file-read ${pyenv_root}/version` is pretty much instantaneous having no output at all (probably doing something wrong). Some for `pyenv version_name`

TL;DR: I can't reproduce it on Linux with very similar zsh and p9k version

**Comment by Zeta611** [comment]
@Syphdias My bad—the huge lag was due to my zsh setting of lazily loading pyenv. However, pyenv segment still lags my prompt five times more than without it. Also, repeating it 1000 takes:
```
( repeat 1000; do; __p9k_build_left_prompt > /dev/null 2>&1; done; )  52.14s user 57.27s system 89% cpu 2:02.36 total
```
But I see that it is incomparably faster—more than 200 times—in your prompt!

**Comment by bhilburn** [comment]
@Zeta611 - Thanks for the testing, and @Syphdias, thank you for trying to reproduce!

Okay, it sounds like we might have a macOS-specific issue, or perhaps an installation-specific issue. @dritter - Do you happen to use Pyenv?

**Comment by romkatv** [comment]
@Zeta611 If you are still using Powerlevel9k, could you check if performance issues can be reproduced with [Powerlevel10k](https://github.com/romkatv/powerlevel10k/#for-powerlevel9k-users)?
